src/modules/form/components/rhf/ControlledSelect.tsx

```diff
@@ -35,6 +36,7 @@ const ControlledSelect: React.FC<Props> = ({
   loading,
   placeholder,
   helperText,
+  afterChange,
```

Comment:
For reference here's a link to our previous convo about implementing an `afterChange` callback in one of these `Controlled` form components, which we ended up scrapping at the time. https://github.com/greenriver/hmis-frontend/pull/850#discussion_r1681533408

TLDR `afterChange` could alternatively be implemented as an effect, but overusing effects when they are not needed can cause extra renders. (https://react.dev/learn/you-might-not-need-an-effect)

Reply:
Sounds good. I think it might be worth making this more general, e.g. `onChange?: (option: PickListOption | null) => void`, so that it's possible to use it for side effects that might depend on the value. I think "onChange" probably clearer than "afterChange" because they are triggered in the same render.